### PR TITLE
chore: copy initial release from internal repository

### PR DESCRIPTION
**Pull request by cawolf**

### CODE REVIEW

**Comment by mmross** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 44
           

Complexity increasing per file
==============================
- Tests/Mock/EventWithException.php  1
- EventListener/FinishControllerSpanSubscriber.php  3
- EventListener/ExceptionLogSubscriber.php  5
- Internal/PersistenceService.php  1
- Tests/EventListener/StartControllerSpanSubscriberTest.php  1
- EventListener/FinishCommandSpanSubscriber.php  1
- Tests/Mock/EventReflectionError.php  1
- Tests/Internal/PersistenceServiceTest.php  1
- Tests/EventListener/FinishCommandSpanSubscriberTest.php  1
- Tests/Internal/UtilityServiceTest.php  1
- EventListener/StartControllerSpanSubscriber.php  1
- Internal/CachedOpentracing.php  2
- Tests/Mock/EventWithThrowable.php  1
- Tests/Mock/EventWithResponse.php  1
- EventListener/StartRootSpanSubscriber.php  2
- DependencyInjection/AbstractOpentracingExtension.php  1
- Tests/EventListener/FinishRootSpanSubscriberTest.php  1
- Tests/Internal/CachedOpentracingTest.php  1
- Tests/Mock/EventWithError.php  1
- Factory/RequestSpanOptionsFactory.php  4
- Internal/UtilityService.php  1
- Tests/Factory/RequestSpanOptionsFactoryTest.php  1
- EventListener/StartCommandSpanSubscriber.php  1
- Tests/EventListener/StartRootSpanSubscriberTest.php  1
- Tests/EventListener/ExceptionLogSubscriberTest.php  1
- EventListener/FinishRootSpanSubscriber.php  2
- Tests/EventListener/StartCommandSpanSubscriberTest.php  1
- Tests/EventListener/FinishControllerSpanSubscriberTest.php  1
- Tests/Mock/MockTracer.php  1
- Service/TracingService.php  3
- Tests/Service/TracingServiceTest.php  1
- Tests/Mock/EventWithResponseAndReflectionError.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/auxmoney/OpentracingBundle-core/pullRequest?prid=4788733&bid=15945802)

**Comment by mmross** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- EventListener/FinishRootSpanSubscriber.php  2
- EventListener/FinishControllerSpanSubscriber.php  3
- Tests/Mock/EventWithResponse.php  1
- Tests/EventListener/FinishRootSpanSubscriberTest.php  1
- Internal/CachedOpentracing.php  2
- EventListener/ExceptionLogSubscriber.php  5
- Tests/Factory/RequestSpanOptionsFactoryTest.php  1
- DependencyInjection/AbstractOpentracingExtension.php  1
- Factory/RequestSpanOptionsFactory.php  4
- EventListener/StartCommandSpanSubscriber.php  1
- Service/TracingService.php  3
- EventListener/FinishCommandSpanSubscriber.php  1
- Tests/EventListener/StartControllerSpanSubscriberTest.php  1
- Tests/EventListener/FinishCommandSpanSubscriberTest.php  1
- Tests/Internal/PersistenceServiceTest.php  1
- Tests/Service/TracingServiceTest.php  1
- Tests/Mock/EventReflectionError.php  1
- Tests/Mock/EventWithResponseAndReflectionError.php  1
- Internal/UtilityService.php  1
- Tests/Internal/CachedOpentracingTest.php  1
- Tests/Internal/UtilityServiceTest.php  1
- Tests/Mock/EventWithError.php  1
- Tests/Mock/EventWithThrowable.php  1
- Internal/PersistenceService.php  1
- Tests/EventListener/StartCommandSpanSubscriberTest.php  1
- Tests/EventListener/FinishControllerSpanSubscriberTest.php  1
- Tests/EventListener/ExceptionLogSubscriberTest.php  1
- EventListener/StartRootSpanSubscriber.php  2
- Tests/EventListener/StartRootSpanSubscriberTest.php  1
- Tests/Mock/MockTracer.php  1
- Tests/Mock/EventWithException.php  1
- EventListener/StartControllerSpanSubscriber.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/auxmoney/OpentracingBundle-core/pullRequest?prid=4788733&bid=15945802)

**Comment by mmross** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- Tests/Mock/EventWithThrowable.php  1
- Tests/EventListener/StartCommandSpanSubscriberTest.php  1
- EventListener/FinishCommandSpanSubscriber.php  1
- Service/TracingService.php  3
- Tests/Mock/EventWithException.php  1
- Tests/Internal/PersistenceServiceTest.php  1
- Tests/EventListener/ExceptionLogSubscriberTest.php  1
- Tests/EventListener/StartControllerSpanSubscriberTest.php  1
- Tests/EventListener/FinishCommandSpanSubscriberTest.php  1
- Tests/Factory/RequestSpanOptionsFactoryTest.php  1
- Internal/UtilityService.php  1
- Tests/Internal/UtilityServiceTest.php  1
- Internal/CachedOpentracing.php  2
- Tests/Service/TracingServiceTest.php  1
- Tests/Mock/MockTracer.php  1
- Tests/EventListener/StartRootSpanSubscriberTest.php  1
- DependencyInjection/AbstractOpentracingExtension.php  1
- Tests/Mock/EventWithError.php  1
- Tests/Mock/EventWithResponse.php  1
- Tests/EventListener/FinishControllerSpanSubscriberTest.php  1
- EventListener/StartRootSpanSubscriber.php  2
- Internal/PersistenceService.php  1
- Tests/Mock/EventWithResponseAndReflectionError.php  1
- Factory/RequestSpanOptionsFactory.php  4
- Tests/Internal/CachedOpentracingTest.php  1
- EventListener/ExceptionLogSubscriber.php  5
- EventListener/FinishControllerSpanSubscriber.php  3
- Tests/Mock/EventReflectionError.php  1
- Tests/EventListener/FinishRootSpanSubscriberTest.php  1
- EventListener/StartCommandSpanSubscriber.php  1
- EventListener/FinishRootSpanSubscriber.php  2
- EventListener/StartControllerSpanSubscriber.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/auxmoney/OpentracingBundle-core/pullRequest?prid=4788733&bid=15945802)

**Comment by mmross** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- EventListener/StartCommandSpanSubscriber.php  1
- EventListener/FinishControllerSpanSubscriber.php  3
- DependencyInjection/AbstractOpentracingExtension.php  1
- Factory/RequestSpanOptionsFactory.php  4
- Internal/PersistenceService.php  1
- EventListener/FinishCommandSpanSubscriber.php  1
- EventListener/FinishRootSpanSubscriber.php  2
- EventListener/ExceptionLogSubscriber.php  5
- EventListener/StartRootSpanSubscriber.php  2
- Service/TracingService.php  3
- Internal/UtilityService.php  1
- EventListener/StartControllerSpanSubscriber.php  1
- Internal/CachedOpentracing.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/auxmoney/OpentracingBundle-core/pullRequest?prid=4788733&bid=15945802)

File: EventListener/ExceptionLogSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    public function __construct(Tracing $tracing)
+    {
+        $this->tracing = $tracing;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/ExceptionLogSubscriber.php]: : array

File: EventListener/FinishCommandSpanSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public function __construct(Tracing $tracing, Persistence $persistence)
+    {
+        $this->tracing = $tracing;
+        $this->persistence = $persistence;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/FinishCommandSpanSubscriber.php]: : array

File: EventListener/FinishControllerSpanSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public function __construct(Tracing $tracing, LoggerInterface $logger)
+    {
+        $this->tracing = $tracing;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/FinishControllerSpanSubscriber.php]: : array

File: EventListener/FinishRootSpanSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public function __construct(Tracing $tracing, Persistence $persistence)
+    {
+        $this->tracing = $tracing;
+        $this->persistence = $persistence;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/FinishRootSpanSubscriber.php]: : array

File: EventListener/StartCommandSpanSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    public function __construct(Tracing $tracing, SpanOptionsFactory $spanOptionsFactory)
+    {
+        $this->tracing = $tracing;
+        $this->spanOptionsFactory = $spanOptionsFactory;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/StartCommandSpanSubscriber.php]: : array

File: EventListener/StartControllerSpanSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    public function __construct(Tracing $tracing)
+    {
+        $this->tracing = $tracing;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/StartControllerSpanSubscriber.php]: s.o.

File: EventListener/StartRootSpanSubscriber.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        SpanOptionsFactory $spanOptionsFactory
+    ) {
+        $this->tracing = $tracing;
+        $this->spanOptionsFactory = $spanOptionsFactory;
+    }
+
+    /**
+     * @return array<string,array>
+     */
+    public static function getSubscribedEvents()
```
Comment (cHeeSaW) [inline EventListener/StartRootSpanSubscriber.php]: s.o.

File: Service/TracingService.php [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+use const OpenTracing\Formats\TEXT_MAP;
+
+final class TracingService implements Tracing
+{
+    private $opentracing;
+    private $logger;
+
+    public function __construct(Opentracing $opentracing, LoggerInterface $logger)
+    {
+        $this->opentracing = $opentracing;
```
Comment (cHeeSaW) [inline Service/TracingService.php]: as discussed: die ganzen getTracerInstance Aufrufe ersetzen

File: Internal/UtilityService.php [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+use OpenTracing\SpanContext;
+use const OpenTracing\Formats\TEXT_MAP;
+
+final class UtilityService implements Utility
+{
+    private $opentracing;
+
+    public function __construct(Opentracing $opentracing)
+    {
+        $this->opentracing = $opentracing;
```
Comment (cHeeSaW) [inline Internal/UtilityService.php]: siehe TracingService

File: Internal/PersistenceService.php [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Auxmoney\OpentracingBundle\Internal;
+
+final class PersistenceService implements Persistence
+{
+    private $opentracing;
+
+    public function __construct(Opentracing $opentracing)
+    {
+        $this->opentracing = $opentracing;
```
Comment (cHeeSaW) [inline Internal/PersistenceService.php]: siehe `TracingService`

File: Service/TracingService.php (hunk 8 above)
Comment (cawolf) [reply Service/TracingService.php]: good idea!

File: Internal/PersistenceService.php (hunk 10 above)
Comment (cawolf) [reply Internal/PersistenceService.php]: see above

**Pull request merged**